### ISSUE TITLE
Create bindings to libxsmm (in a separate package)

### ISSUE THREAD

**Issue by Mikolaj** [Mikolaj/horde-ad#41]
See https://github.com/Mikolaj/horde-ad/issues/9#issuecomment-1086211063.'

To test the bindings, implement convolution in horde-ad tests using them.

**Comment by Mikolaj** [comment]
The library also supposedly implements LSTM. If that implementation is adequate and with pure enough API, let's create some example neural networks with that. If not and if no libraries offer a good alternative, let's implement our own. Two good blog posts about LSTM: https://jasdeep06.github.io/posts/Understanding-LSTM-in-Tensorflow-MNIST and https://colah.github.io/posts/2015-08-Understanding-LSTMs.

**Comment by Mikolaj** [comment]
Actually, I'm not even sure backpropagating through all (popular) variants of LSTM and GRU can be modelled by expressing the whole neural net, including the LSTM, as a single function and taking its gradients. I'm seeing hints that variants with "hard attention" can't, as opposed to those with "soft attention". That's a theory question related to non-differentiable functions (nns with the bad LSTM are supposedly not differentiable despite their various components being so). CC @fkrawiec.
